Mage.Sets/src/mage/cards/b/BlitzballStadium.java

```diff
@@ -0,0 +1,95 @@
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.CompositeCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.UntapSourceEffect;
+import mage.abilities.effects.common.combat.CantBeBlockedTargetEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.abilities.effects.common.counter.AddCountersTargetEffect;
+import mage.abilities.hint.ValueHint;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCreaturePermanent;
+import mage.target.targetadjustment.XTargetsCountAdjuster;
+
+/**
+ * @author balazskristof
+ */
+public final class BlitzballStadium extends CardImpl {
+
+    public static final ValueHint hint = new ValueHint("Kind of counters on it", BlitzballStadiumValue.instance);
+
+    public BlitzballStadium(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{X}{U}");
+
+        // When this artifact enters, support X.
+        Ability supportAbility = new EntersBattlefieldTriggeredAbility(new AddCountersTargetEffect(CounterType.P1P1.createInstance())
+                .setText("support X. <i>(Put a +1/+1 counter on each of up to X target creatures.)</i>")
+        );
+        supportAbility.addTarget(new TargetCreaturePermanent(0, 1));
+        supportAbility.setTargetAdjuster(new XTargetsCountAdjuster());
+        this.addAbility(supportAbility);
+
+        // Go for the Goal! -- {3},{T}: Until end of turn, target creature gains "Whenever this creature deals combat damage to a player, draw a card for each kind of counter on it" and it can't be blocked this turn.
+        Ability ability = new SimpleActivatedAbility(
+                new GainAbilityTargetEffect(
+                        new DealsCombatDamageToAPlayerTriggeredAbility(
+                                new DrawCardSourceControllerEffect(BlitzballStadiumValue.instance)
+                                    .setText("draw a card for each kind of counter on it"),
+                                false
+                        ).addHint(hint)
+                ).setText("Until end of turn, target creature gains \"Whenever this creature deals combat damage to a player, draw a card for each kind of counter on it\""),
+                new CompositeCost(new ManaCostsImpl<>("{3}"), new TapSourceCost(), "{3},{T}")
+        );
+        ability.addEffect(new CantBeBlockedTargetEffect().setText("it can't be blocked this turn").concatBy("and"));
```

Comment:
this is a minor nitpick and something you don't need to change, but you can combine `setText` and `concatBy` into a single call since `concatBy` mostly exists to prevent the need to call `setText`